## Probability of cancer given test results and population of people who have cancer

<img src='https://i.imgur.com/EPaOezY.png' align='left'>

In [23]:
prob_cancer = 0.1
# high senstivity 
prob_pos_given_cancer = 0.9
# low specitivity
prob_neg_given_no_cancer = 0.5

### Negative Result

In [47]:
prob_cancer_given_neg = prob_cancer * (1-prob_pos_given_cancer)
prob_cancer_given_neg

0.009999999999999998

In [33]:
prob_not_cancer_given_neg = (1-prob_cancer) * prob_neg_given_no_cancer
prob_not_cancer_given_neg

0.45

In [38]:
prob_neg_result = prob_cancer_given_neg + prob_not_cancer_given_neg
prob_neg_result

0.46

In [39]:
norm_prob_cancer_given_neg = prob_cancer_given_neg/prob_neg_result
norm_prob_cancer_given_neg

0.021739130434782605

In [40]:
norm_prob_not_cancer_given_neg = prob_not_cancer_given_neg/prob_neg_result
norm_prob_not_cancer_given_neg

0.9782608695652174

In [41]:
sum_to_1_check = norm_prob_cancer_given_neg + norm_prob_not_cancer_given_neg
sum_to_1_check

1.0

### Positive Result

$ P(C \mid P ) = \frac{P(C) \cdot P(P \mid C)}{P(P)} $

In [55]:
# two ways, to get a positive result, correct positive (sensitivity) or a false positive (1 - specitivity)
prob_pos_result = prob_cancer * prob_pos_given_cancer + (1-prob_cancer) * (1-prob_neg_given_no_cancer)

# probability of cancer and test was correct divide by chance of a positive result
prob_C_given_P = prob_cancer * prob_pos_given_cancer/(prob_pos_result)
prob_C_given_P

0.16666666666666669

In [57]:
prob_not_C_given_P = (1-prob_cancer) * (1-prob_neg_given_no_cancer) / prob_pos_result
prob_not_C_given_P

0.8333333333333333

Our chance of having cancer given a positive result only increases from 10% to 16%  
  
Our chance of not having cancer given a negative result increases from 90% to 98%